modules/apps/headless/headless-admin-address/headless-admin-address-impl/src/main/java/com/liferay/headless/admin/address/internal/resource/v1_0/CountryResourceImpl.java

```diff
@@ -141,8 +143,10 @@ public Country putCountry(Long countryId, Country country)
 				GetterUtil.getBoolean(country.getShippingAllowed(), true),
 				GetterUtil.getBoolean(country.getSubjectToVAT()));
 
-		_countryLocalService.updateCountryLocalizations(
-			serviceBuilderCountry, country.getTitle_i18n());
+		if (country.getTitle_i18n() != null) {
+			_countryLocalService.updateCountryLocalizations(
+				serviceBuilderCountry, country.getTitle_i18n());
+		}
```

Comment:
@ChrisKian does this not break the PUT expectation that the resource data is fully replaced with the new data?
I am thinking when the there is already a localization prior to the PUT.

Reply:
If we had an existing country with existing localized titles, and we updated it through the PUT method, here are the three possibilities in doing so:

1. We add `title_i18n` with non-blank data to the request, and the new title data overwrites the existing data.
2. We add `title_i18n` with blank data to the request, and the existing title(s) are overwritten to blank values (allowed, because we don't require title data).
3. We completely exclude the `title_i18n` attribute in the request, so we don't perform any updates to the existing title localizations.  (I am under the impression this is acceptable, but I could be wrong)

I didn't see anywhere explicitly requiring the `title_i18n` attribute to be set, so I assumed it was optional.  If this is not the case, then my fix is incorrect, and there are some tests which need to be fixed as well.  Please let me know how we should proceed, thanks!

Reply:
Hi @ChrisKian . My expectation is that [3] should be the behaviour of PATCH . 

When any optional attribute is excluded in a PUT request, their default value should be used, which I think is a blank localization in this case.

Have you checked other headless resources that support these verbs and have optional localized fields?

Reply:
Hey Stian, you are correct.  I've sent a new PR which should fully resolve the original issue and the one you pointed out here.  https://github.com/liferay-appsec/liferay-portal/pull/1425